functional/platform/nspawn.go

```diff
@@ -128,6 +128,46 @@ func (nc *nspawnCluster) FleetctlWithInput(m Member, input string, args ...strin
 	return util.RunFleetctlWithInput(input, args...)
 }
 
+// WaitForNUnits runs fleetctl list-units to verify the actual number of units
+// matched with the given expected number. It periodically runs list-units
+// waiting until list-units actually shows the expected units.
+func (nc *nspawnCluster) WaitForNUnits(m Member, expectedUnits int) (map[string][]util.UnitState, error) {
+	var nUnits int
+	retStates := make(map[string][]util.UnitState)
+	checkListUnits := func() bool {
+		outListUnits, _, err := nc.Fleetctl(m, "list-units", "--no-legend", "--full", "--fields", "unit,active,machine")
+		if err != nil {
+			return false
+		}
+		if expectedUnits == 0 && len(outListUnits) == 0 {
+			return true
+		}
```

Comment:
@antrik 
> Note that ParseUnitStates() implicitly filters out an empty line

Yeah, I just got cold feet. Actually I would not like to rely on such implicit behavior. Anyway I removed checking for empty line, and wrote a comment about that.